### PR TITLE
Added duplicate functionality from the gig detail page. This basicall…

### PR DESCRIPTION
**Pull request by aaronoppenheimer**
…y uses the 'create' path, but populates the edit form from the gig we're duplicating.

Based on 'datepicker2' branch which should be merged first.

### CODE REVIEW

**Comment by aaronoppenheimer** [comment]
To clarify, this is "duplicate this gig by pressing the 'duplicate' button" not some functionality that duplicates something else.

File: gig/views.py [hunk 1]
```diff
@@ -114,11 +116,33 @@ def form_valid(self, form):
...
+        # ...but replace the title with a 'copy of'
+        kwargs['initial']['title'] = f'Copy of {kwargs["initial"]["title"]}'
+        return kwargs
+
+    def get_band_from_kwargs(self, **kwargs):
+        # didn't have the band from the request args, so pull it from the gig
+        try:
+            return Gig.objects.get(id=self.kwargs['pk']).band
+        except Gig.DoesNotExist:
+            raise Http404('No gig found matching the query')
```
Comment (rschroll) [inline gig/views.py:140]: Perhaps a place to use `get_object_or_404`.

File: templates/gig/gig_detail.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
                                 <a class="btn btn-primary btn-sm" href="/gig/{{ gig.id }}/update">{% trans "Edit" %}</a>
                             {% endif %}
                             {% if user_can_create %}
-                                <a class="btn btn-primary btn-sm" href="gig_edit.html?gk={{ gig.id }}&dupe=1">{% trans "Duplicate" %}</a>
+                                <a class="btn btn-primary btn-sm" href="/gig/{{ gig.id }}/duplicate">{% trans "Duplicate" %}</a>
```
Comment (rschroll) [inline templates/gig/gig_detail.html:31]: I'd recommend the {% url %} function.

File: gig/views.py (hunk 1 above)
Comment (aaronoppenheimer) [reply gig/views.py:140]: "perhaps" - you're too nice

**Pull request merged**